docs/testing/services-custom/dynamodb-enhanced/enhanced-client-testing.md

```diff
@@ -0,0 +1,59 @@
+# DynamoDB Enhanced Client Test Checklist
+
+Below is the checklist required for merging DynamoDB Enhanced Client test changes.
+Please complete this checklist when making any changes to Enhanced DDB.
+
+**Instructions:**
+- Mark completed tests with [x] (e.g., [x] instead of [ ])
+- For tests not completed, provide a reason in the "Comments" column
+- If your code changes are covered by existing tests, reference the test class and method name in the Comments column (e.g., `TableSchemaTest.testFromBean()`)
+- Please copy the markdown into your PR and update it in Testing section
+- [Reference for CRUD Operations test](https://github.com/aws/aws-sdk-java-v2/blob/master/services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/BasicCrudTest.java)
+- [Reference for Data Types and Null Handling](https://github.com/aws/aws-sdk-java-v2/blob/master/services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/document/EnhancedDocumentTestData.java) 
+
+```md
+## Test Coverage Checklist
+
+| Scenario | Done | Comments if Not Done |
+|---------|:----:|---------------------|
+| **1. Different TableSchema Creation Methods** | | |
+| a. TableSchema.fromBean(Customer.class) | [ ] | |
+| b. TableSchema.fromImmutableClass(Customer.class) for immutable classes | [ ] | |
+| c. TableSchema.documentSchemaBuilder().build() | [ ] | |
+| d. StaticTableSchema.builder(Customer.class) | [ ] | |
+| **2. Nesting of Different TableSchema Types** | | |
+| a. @DynamoDbBean with nested @DynamoDbBean as NonNull | [ ] | |
```

Comment:
What does NonNull mean?

Reply:
As in null reference , updated to make it clear